plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadDumpAction.java

```diff
@@ -108,13 +118,47 @@ public CpsThreadDump getThreadDump() {
 
         @Override public void addContents(Container container) {
             container.add(new Content("nodes/master/pipeline-thread-dump.txt") {
-                @Override public void writeTo(OutputStream outputStream) throws IOException {
+                @Override public void writeTo(OutputStream outputStream) {
                     PrintWriter pw = new PrintWriter(new OutputStreamWriter(outputStream, StandardCharsets.UTF_8));
                     for (FlowExecution flow : FlowExecutionList.get()) {
                         if (flow instanceof CpsFlowExecution) {
-                            pw.println("Build: " + flow.getOwner().getExecutable());
-                            ((CpsFlowExecution) flow).getThreadDump().print(pw);
-                            pw.println("Approximate graph size: " + ((CpsFlowExecution) flow).approximateNodeCount());
+                            Queue.Executable ownerExec;
+                            try {
+                                ownerExec = flow.getOwner().getExecutable();
+                            } catch (IOException e) {
+                                pw.println("No data available for " + flow);
+                                Functions.printStackTrace(e, pw);
+                                pw.println();
+                                continue;
+                            }
+                            pw.println("Build: " + ownerExec);
+                            if (ownerExec instanceof Run<?, ?>) {
+                                var run = (Run<?, ?>) ownerExec;
+                                var started = Instant.ofEpochMilli(run.getStartTimeInMillis());
+                                pw.println("Started: " + started);
+                                var duration = Duration.between(started, Instant.now());
+                                pw.print("Duration: " + duration);
```

Comment:
Not very legible; consider `Util.getTimeSpanString` (IIRC).

Reply:
Yes, this is subjective (personally I like ISO 8601 in contexts like this). Mainly I did not use `getTimeSpanString` to avoid localization.